node/src/components/linear_chain_sync/operations.rs

```diff
@@ -0,0 +1,288 @@
+use std::{fmt::Debug, time::Duration};
+
+use rand::seq::SliceRandom;
+use std::collections::BTreeMap;
+use tracing::{trace, warn};
+
+use casper_execution_engine::{
+    shared::{newtypes::Blake2bHash, stored_value::StoredValue},
+    storage::trie::Trie,
+};
+use casper_types::{EraId, Key, PublicKey, U512};
+
+use crate::{
+    components::{
+        consensus::era_supervisor,
+        fetcher::{FetchedData, FetcherError},
+        linear_chain_sync::error::LinearChainSyncError,
+    },
+    effect::{
+        requests::{ContractRuntimeRequest, FetcherRequest, NetworkInfoRequest},
+        EffectBuilder,
+    },
+    types::{BlockHash, BlockHeader, BlockHeaderWithMetadata, Chainspec, Item},
+};
+
+const TIMEOUT_DURATION: Duration = Duration::from_millis(100);
+
+/// Get the currently connected peers from the networking component and shuffle them
+async fn get_and_shuffle_network_peers<REv, I>(effect_builder: EffectBuilder<REv>) -> Vec<I>
+where
+    REv: From<NetworkInfoRequest<I>>,
+    I: Send + 'static,
+{
+    let mut vector_of_peers: Vec<I> = effect_builder
+        .network_peers::<I>()
+        .await
+        .into_iter()
+        .map(|(peer, _)| peer)
+        .collect();
+    let mut rng = rand::thread_rng();
```

Comment:
The current tests for this code are non-deterministic, since they are somewhat high level.

It is a bit complicated to thread it in, but if it's a show stopper I can ticket carrying it through the code.

Reply:
Can't we just call `NodeRng::new()` here?

Reply:
Fixed in c379588f655a74483cdc8dac7f5449380d5f038d